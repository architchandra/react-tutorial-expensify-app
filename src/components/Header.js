import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';



const Header = (props) => (
  <header>
    <Link to="/dashboard" exact={ true }>
      <h1>
        Expensify
      </h1>
    </Link>
    <button onClick={props.startLogout}>
      Logout
    </button>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout()),
});



export { Header };
export default connect(undefined, mapDispatchToProps)(Header);
