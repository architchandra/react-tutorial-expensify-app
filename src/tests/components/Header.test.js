import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../../components/Header';




test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
