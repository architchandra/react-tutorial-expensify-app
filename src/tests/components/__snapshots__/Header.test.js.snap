// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="navitem-active"
        ariaCurrent="true"
        exact={true}
        to="/dashboard"
      >
        Dashboard
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="navitem-active"
        ariaCurrent="true"
        to="/create"
      >
        Create
      </NavLink>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Logout
      </button>
    </li>
  </ul>
</header>
`;
